loopy/target/c/codegen/expression.py

```diff
@@ -99,13 +99,7 @@ def find_array(self, expr):
         return ary
 
     def wrap_in_typecast(self, actual_type, needed_dtype, s):
-        if (actual_type.is_complex() and needed_dtype.is_complex()
-                and actual_type != needed_dtype):
-            return var("%s_cast" % self.complex_type_name(needed_dtype))(s)
-        elif not actual_type.is_complex() and needed_dtype.is_complex():
-            return var("%s_fromreal" % self.complex_type_name(needed_dtype))(s)
-        else:
-            return s
+        return s
```

Comment:
This worries me, since it turns a function into a no-op, doing something eminently different than what it says on the tin. If the function is no longer called/serves no purpose, maybe remove it?

Reply:
IIUC, this interface is provided by the base class so that targets that do not have implicit type conversions could just override this one method. For example. `double complex z = 1.0;` in legal C-target but  `cdouble_t x = 1.0;` would be incorrect in the pyopencl target.

Just getting rid of this interface would lead to heavy rewriting for the sub-class' mapper at which point deriving would result in very little returns.

Reply:
Ah, nvm. This is just moved to `ExpressionToPyOpenCLCExpressionMapper`.